### PR TITLE
Remove leading comments from terminal submission (#1244)

### PR DESCRIPTION
**Pull request by gowerc**
# What problem did you solve?

Code removes leading comments from code submitted to terminal  (issue #1244)

## (If you have)Screenshot

[remove_comments_screencast.webm](https://user-images.githubusercontent.com/8447209/198898477-84240d2a-f9c1-4579-83f8-a23f9c9c7efa.webm)

### CODE REVIEW

**Comment by gowerc** [comment]
I’m now wondering if this instead should be moved to within expandselction? This current implementation would also strip leading comments off of explicitly highlighted chunks of code, though maybe that’s not an issue, I’m not too sure?

File: src/selection.ts [hunk 1]
```diff
@@ -72,7 +72,9 @@ export function getSelection(): RSelection | undefined {
         selection.range = new Range(newStart, newEnd);
     }
 
-    selection.selectedText = currentDocument.getText(selection.range).trim();
+    selection.selectedText = removeLeadingComments(
```
Comment (renkun-ken) [inline src/selection.ts:75]: Just adding a setting to opt-in this behavior is ok for me.

**Review by renkun-ken — APPROVED** [review]
LGTM Thanks!

**Comment by gowerc** [comment]
Just wanedt to apologies for not getting around to finishing this off myself, we just had a baby daughter which is consuming all my spare time at the moment 😄

**Pull request merged**